src/hotspot/share/gc/g1/heapRegion.inline.hpp

```diff
@@ -79,8 +79,30 @@ inline HeapWord* HeapRegion::par_allocate_impl(size_t min_word_size,
   } while (true);
 }
 
-inline HeapWord* HeapRegion::block_start(const void* p) {
-  return _bot_part.block_start(p);
+inline HeapWord* HeapRegion::forward_to_block_containing_addr(HeapWord* q, HeapWord* n,
+                                                              const void* addr) const {
+  while (n <= addr) {
+    // When addr is not covered by the block starting at q we need to
+    // step forward until we find the correct block. With the BOT
+    // being precise, we should never have to step through more than
+    // a single card.
+    _bot_part.assert_same_bot_entry(n, addr);
```

Comment:
I think `assert(!is_crossing_card_boundary(n, addr)` matches the comment better.